src/ruby/qps/histogram.rb

```diff
@@ -70,4 +70,27 @@ def count
   def contents
     @buckets
   end
+
+  def set_minimum(value)
+    if value < @min_seen
+      @min_seen = value
+    end
+  end
+  def set_maximum(value)
+    if value > @max_seen
+      @max_seen = value
+    end
+  end
+  def add_sum(sum_set)
+    @sum += sum_set
+  end
+  def add_sum_of_squares(soq_set)
+    @sum_of_squares += soq_set
+  end
+  def add_count(count_set)
+    @count += count_set
+  end
+  def add_contents(contents_set)
+    @buckets = @buckets.map.with_index{ |m,i| m + contents_set[i].to_i }
```

Comment:
You should collapse all of set_minimum, set_maximum, add_sum, add_sum_of_squares, and add_count into this function. Just pass in the whole `lat`, not just `lat.bucket.to_a` and then do all the operations together since there's never a need to do any one of those operations without doing all of them. Then you can eliminate all the other unneeded functions.

Reply:
Done.